pkg/controller/oneagent/oneagent_controller.go

```diff
@@ -454,15 +451,60 @@ func newPodSpecForCR(instance *dynatracev1alpha1.OneAgent) corev1.PodSpec {
 				},
 			},
 		},
-		Volumes: []corev1.Volume{{
+		Volumes: prepareVolumes(instance),
+	}
+}
+
+func prepareVolumes(instance *dynatracev1alpha1.OneAgent) []corev1.Volume {
+	volumes := []corev1.Volume{
+		{
 			Name: "host-root",
 			VolumeSource: corev1.VolumeSource{
 				HostPath: &corev1.HostPathVolumeSource{
 					Path: "/",
 				},
 			},
-		}},
+		},
 	}
+
+	if instance.Spec.TrustedCAs != "" {
+		volumes = append(volumes, corev1.Volume{
+			Name: "certs",
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: instance.Spec.TrustedCAs,
+					},
+					Items: []corev1.KeyToPath{
+						{
+							Key:  "certs",
+							Path: "certs.pem",
+						},
+					},
+				},
+			},
+		})
+	}
+
+	return volumes
+}
+
+func prepareVolumeMounts(instance *dynatracev1alpha1.OneAgent) []corev1.VolumeMount {
+	volumeMounts := []corev1.VolumeMount{
+		{
+			Name:      "host-root",
+			MountPath: "/mnt/root",
+		},
+	}
+
+	if instance.Spec.TrustedCAs != "" {
+		volumeMounts = append(volumeMounts, corev1.VolumeMount{
+			Name:      "certs",
+			MountPath: "/mnt/custom_certificates",
```

Comment:
As discussed internally,
```suggestion
			MountPath: "/mnt/dynatrace/certs",
```